x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/common/task/siem_migrations_task_evaluator.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { EvaluationResult } from 'langsmith/evaluation';
+import type { RunnableConfig } from '@langchain/core/runnables';
+import type { Run, Example } from 'langsmith/schemas';
+import { evaluate } from 'langsmith/evaluation';
+import { isLangSmithEnabled } from '@kbn/langchain/server/tracers/langsmith';
+import { Client } from 'langsmith';
+import type { Logger } from '@kbn/logging';
+import type { LangSmithEvaluationOptions } from '../../../../../common/siem_migrations/model/common.gen';
+import type { SiemMigrationTaskRunner } from './siem_migrations_task_runner';
+import type { MigrationDocument, ItemDocument, SiemMigrationsClientDependencies } from '../types';
+
+export interface EvaluateParams<C extends object = {}> {
+  connectorId: string;
+  langsmithOptions: LangSmithEvaluationOptions;
+  invocationConfig?: RunnableConfig<C>;
+}
+
+export type Evaluator = (args: { run: Run; example: Example }) => EvaluationResult;
+type CustomEvaluatorResult = Omit<EvaluationResult, 'key'>;
+export type CustomEvaluator = (args: { run: Run; example: Example }) => CustomEvaluatorResult;
```

Comment:
@semd , what is the difference between `CustomEvaluator` and `Evaluator`? They seem same.